core/lib/zksync_core/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -18,19 +20,26 @@ mod metrics;
 #[cfg(test)]
 mod tests;
 
-/// This component keeps track of the median base_fee from the last `max_base_fee_samples` blocks.
+/// This component keeps track of the median `base_fee` from the last `max_base_fee_samples` blocks
+/// and of the median `blob_base_fee` from the last `max_blob_base_fee_sample` blocks.
 /// It is used to adjust the base_fee of transactions sent to L1.
 #[derive(Debug)]
 pub struct GasAdjuster {
-    pub(super) statistics: GasStatistics,
+    pub(super) base_fee_statistics: GasStatistics<u64>,
```

Comment:
Do we really need to keep it `u64`? With [custom base token](https://github.com/matter-labs/zksync-era/pull/996) the gas prices here may also become too big, so probably instead of making `GasStatistics` generic, it'd be better to simply make it always work with `U256`.

Reply:
@popzxc I'm not sure about the implications, how overflow should be handled, etc, and don't want it to be a part of this PR. It was pretty easy to make `GasStatistics` generic, so in future we can always rollback if needed.

Reply:
Sure, let's keep it out of scope then.